### ISSUE TITLE
Memory Leaks when compile by translator

### ISSUE THREAD

**Issue by colderlord** [bridgedotnet/Bridge#3936]
i tried to optimize compilation in issues [3805](https://github.com/bridgedotnet/Bridge/issues/3824) and [3824](https://github.com/bridgedotnet/Bridge/issues/3824) , but it cause memory leaks.
i delete all code that i've wrote.

**Comment by fabriciomurta** [comment]
How exactly did you notice the leak? Perhaps some bit in the dependencies is changing between builds and they get added again? Or maybe just always calling `new ReferenceInfo()`, so replacing `GetOrAdd` with something that returns a _cached_ instance of `ReferenceInfo`?

Anyways, thanks for your honesty on coming back at the contribution for finding issues with it. We really appreciate that! During the tests no noticeable leak was found, so it shouldn't be affecting too much in a general use case (where the compiler/instance is closed and the leaking resources, freed).

**Comment by fabriciomurta** [comment]
As the benefits we could touch with the implementation were not noticeable on what we had to test with, we couldn't really reproduce the memory leaks either, yet coming back on the implementation won't hurt anyone else. So we just went ahead and accepted the pull request. By the way, there was a class (`ReferenceInfo`) that was not used anywhere else (introduced by the related issues), and was removed as well.

Bridge without the references caching code will be available next release, according to this issue's milestone.
